deeplake/enterprise/dataloader.py

```diff
@@ -689,6 +689,16 @@ def _get_suboptimal_thread_count(self) -> Optional[int]:
             return num_suboptimal_threads
         return self._num_threads
 
+    def __is_upcast_needed(self, dataset, tensors) -> bool:
+        for tensor_name in tensors:
+            tensor = dataset._get_tensor_from_root(tensor_name)
+            dtype = tensor.dtype
+            if dtype is None:
+                return True
+            if isinstance(dtype, (np.uint16, np.uint32, np.uint64)):
```

Comment:
How did you come up with these dtypes? np.uint16, np.uint32, np.uint64, I mean why not np.uint8 etc.?

Reply:
here upcasting is only needed if the end framework is the pytorch one. So those are the types which are not supported there. Below is the link to their supported dtypes

https://pytorch.org/docs/stable/tensors.html

Reply:
I think I am missing something, do you mean to say that if I send the array with dtype np.uint16 the PyTorch's dataloader will fail? However, if I send it with dtype np.int16 it will automatically cast it to torch.int16?

Reply:
oh, I was able to understand the proble